catroid/src/main/java/org/catrobat/catroid/content/actions/GoNStepsBackAction.java

```diff
@@ -53,42 +53,20 @@ protected void update(float delta) {
 		if (stepsValue.intValue() > 0 && (zPosition - stepsValue.intValue()) < Constants.Z_INDEX_FIRST_SPRITE) {
 			sprite.look.setZIndex(Constants.Z_INDEX_FIRST_SPRITE);
 		} else if (stepsValue.intValue() < 0 && (zPosition - stepsValue.intValue()) < zPosition) {
-			toFront();
+			toFront(delta);
 		} else {
 			goNStepsBack(stepsValue.intValue());
 		}
 	}
 
-	private void toFront() {
-
-		List<Sprite> spriteList = ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones();
-		int actualSpriteZIndex = sprite.look.getZIndex();
-
-		for (int i = 0; i < spriteList.size(); i++) {
-			if (spriteList.get(i).look.getZIndex() > actualSpriteZIndex) {
-				spriteList.get(i).look.setZIndex(spriteList.get(i).look.getZIndex() - 1);
-			}
-		}
-		sprite.look.setZIndex(spriteList.size() - 1);
+	private void toFront(float delta) {
+		ComeToFrontAction mComeToFrontAction = (ComeToFrontAction) sprite.getActionFactory().createComeToFrontAction(sprite);
+		mComeToFrontAction.update(delta);
```

Comment:
Please do not prefix variables with a 'm'. Neither member variables nor local variables.
I would call the method `act` instead of `update` because the Stage does also call `act` and you can remove the casting. The execution sequence of `act` is `begin`, `update`, `end`.